app/helpers/katello/providers_helper.rb

```diff
@@ -20,11 +20,8 @@ def redhat_repo_tab(provider, tab_id)
       tabs = {}.with_indifferent_access
       redhat_repo_tabs.each { |tab| tabs[tab[:id]] = tab }
 
-      product_content = provider.organization.filtered_product_content
-
-      provider.products.each do |product|
-        selected_contents = product_content.select { |pc| pc.product_id == product.id }.select(&:displayable?)
-        selected_contents.each do |prod_content|
+      provider.products.includes(:displayable_product_contents).each do |product|
+        product.displayable_product_contents.each do |prod_content|
```

Comment:
I think this is the reason that the ui is 2x as slow.  Adding an index to 'content_type' seems to speed it up.  See if it does it for you as well.  

I'm seeing some weirdness around the requests not being returned to the browser for an additional ~20 seconds after rails is done with it.  Trying to figure that out (i don't see it on master)

Reply:
Alright, i figured it out.  For some reason the 'bullet' gem is very very slow when you've loaded a lot of objects form the db.  In this instance (and my test db) it was loading ~4K objects from the db, only taking a few seconds.  Bullet then took ~25 seconds to figure out if the eager loaded was needed or not.

This won't happen in production because 'bullet' doesnt exist there. Slowing down developement by that much is not great, but since this page should be gone 'soon', I think i'm okay with it.  I don't think there are any alternatives aside from reducing the amount of objects loaded via storing more attributes in the db (which i don't know is worth it).

Reply:
Disabling bullet with an around_action call worked very nicely!